graphql/resolvers.go

```diff
@@ -3350,6 +3352,20 @@ func (r *taskResolver) BaseTask(ctx context.Context, obj *restModel.APITask) (*r
 	return apiTask, nil
 }
 
+func (r *taskResolver) TaskDuration(ctx context.Context, obj *restModel.APITask) (*restModel.APIDuration, error) {
+	// TaskDuration is sometimes added via aggregation when Task is resolved via GetTasksByVersion.
+	if obj.TaskDuration != 0 {
+		return &obj.TaskDuration, nil
+	}
+	// If status is running, calculate the time elapsed since it started.
+	if utility.FromStringPtr(obj.DisplayStatus) == evergreen.TaskStarted {
+		timeSinceStart := restModel.NewAPIDuration(time.Since(utility.FromTimePtr(obj.StartTime)))
+		return &timeSinceStart, nil
+	}
+	// If any other status, return timeTaken.
+	return &obj.TimeTaken, nil
```

Comment:
Would it make sense to return nil in the case that the task hasn't started / time taken is the zero time? (Not sure how this will be used but)

Reply:
update; removed `TaskDuration` resolver entirely